src/ScoreCard.js

```diff
@@ -0,0 +1,125 @@
+var ScoreCard = function(name) {
+
+  this.name = name;
+  this.score = [[]];
+  this.frameNo = 0;
+  this.playTurn = false;
+  var currFrame = [];
+  var frameBefore = [];
+  var frameBeforeBefore = [];
+
+
+  ScoreCard.prototype.currentScore = function() {
+
+    var total = 0;
+    for(var frame=0, n=this.score.length; frame < n; frame++) {
+      var subTotal = 0;
+      for(var i=0, x=this.score[frame].length; i < x; i++) {
+        subTotal += this.score[frame][i];
+      }
+      total += subTotal;
+    }
+
+    return total;
+
+  };
+
+  ScoreCard.prototype.addScore = function(rollScore) {
+    currFrame = this.score[this.frameNo];
+    if (this._isAnyFrameAfterThe1st()) {frameBefore = this.score[this.frameNo-1]};
+    if (this._isAnyFrameAfterThe2nd()) {frameBeforeBefore = this.score[this.frameNo-2]};
+    
+    this._scoreCheck(rollScore);
+
+    if (this._isAnyFrameUpToThe10th() ) {
+      currFrame.push(rollScore);
+      if (this._frameBeforeIsASpare() && this._wasThe1stRollOfFrame() ) {frameBefore.push(rollScore)};
+      if (this._frameBeforeIsASrike() ) {frameBefore.push(rollScore)};
+      if (this._previousTwoRollsAreSrikes() && this._wasThe1stRollOfFrame() ) {frameBeforeBefore.push(rollScore)};
+    } else {
+      if (this._frameBeforeIsASpare() || this._frameBeforeIsASrike() ) {frameBefore.push(rollScore)};
+      if (this._previousTwoRollsAreSrikes() && this._isThe11thFrame() ) {frameBeforeBefore.push(rollScore)};
+      if (this._frameBeforeBeforeIsASrike && this._isThe12thFrame() ) {frameBeforeBefore.push(rollScore)};
+   };
+
+    if (rollScore === 10 || this._wasThe2ndRollOfFrame() || (this.frameNo === 10) ) {
+      this.frameNo += 1;
+      this.playTurn = false;
+      this.score[this.frameNo] = [];
+    };
+
+    if (this._gameIsFinished) {this._alertGameIsFinished()};
+
+  };
+  
```

Comment:
This function is way too big, illegible, and has too many responsibilities. Refactor into concise functions that each have a single responsibility.